mmpose/apis/train.py

```diff
@@ -46,22 +48,39 @@ def train_model(model,
         build_dataloader(ds, **dataloader_setting) for ds in dataset
     ]
 
+    # determine wether use adversarial training precess or not
+    use_adverserial_train = cfg.get('use_adversarial_train', False)
+
     # put model on gpus
     if distributed:
         find_unused_parameters = cfg.get('find_unused_parameters', True)
         # Sets the `find_unused_parameters` parameter in
         # torch.nn.parallel.DistributedDataParallel
-        model = MMDistributedDataParallel(
-            model.cuda(),
-            device_ids=[torch.cuda.current_device()],
-            broadcast_buffers=False,
-            find_unused_parameters=find_unused_parameters)
+
+        if use_adverserial_train:
+            # Use DistributedDataParallelWrapper for adversarial training
+            model = DistributedDataParallelWrapper(
+                model,
+                device_ids=[torch.cuda.current_device()],
+                broadcast_buffers=False,
+                find_unused_parameters=find_unused_parameters)
+        else:
+            model = MMDistributedDataParallel(
+                model.cuda(),
+                device_ids=[torch.cuda.current_device()],
+                broadcast_buffers=False,
+                find_unused_parameters=find_unused_parameters)
     else:
         model = MMDataParallel(
             model.cuda(cfg.gpu_ids[0]), device_ids=cfg.gpu_ids)
 
     # build runner
-    optimizer = build_optimizer(model, cfg.optimizer)
+    if use_adverserial_train:
+        # build multiple optimizers for generator and discriminator
+        optimizer = build_optimizers(model, cfg.optimizer)
+    else:
+        optimizer = build_optimizer(model, cfg.optimizer)
```

Comment:
can `build_optimizers` fall back on `build_optimizer`?

Reply:
In adversarial training, we need to build multiple optimizers, so I use the function `build_optimizers` in mmediting.  Maybe we can use `build_optimizers` to replace `build_optimizer`.